src/pollers/gateway.rs

```diff
@@ -30,16 +30,21 @@ use tonic::{
     codegen::{InterceptedService, Service},
     service::Interceptor,
     transport::{Certificate, Channel, Endpoint, Identity},
-    Status,
+    IntoRequest, Status,
 };
 use tower::ServiceBuilder;
 use url::Url;
 use uuid::Uuid;
 
 #[cfg(test)]
 use futures::Future;
+use std::collections::HashMap;
+use std::str::FromStr;
+use tonic::metadata::{MetadataKey, MetadataValue};
 
 static LONG_POLL_METHOD_NAMES: [&str; 2] = ["PollWorkflowTaskQueue", "PollActivityTaskQueue"];
+const LONG_POLL_TIMEOUT: Duration = Duration::from_secs(60);
```

Comment:
I would make it a little big longer than 60 to avoid race condition with the service returning the call after 60 seconds. May be 61 second?

Reply:
Done. Went with 62 because even numbers are nice :)